src/utils/settings.ts

```diff
@@ -0,0 +1,39 @@
+import { DEFAULT_GAS_PRICE } from 'config'
+
+export interface UserSettings {
+  gasPrice: string
+}
+
+export const VERSION = 1
+
+export const GAS_SETTINGS = {
+  default: DEFAULT_GAS_PRICE, // 5
+  fast: '10000000000', // 10
+  reallyfast: '15000000000', // 15
+}
+
+export const getDefaultSettings = (): UserSettings => ({
+  gasPrice: GAS_SETTINGS.default,
+})
+
+export const getStorageKey = (account: string) => {
+  return `pancakeswap_settings_${account}_${VERSION}`
+}
+
+export const getSettings = (account: string): UserSettings => {
+  try {
+    const settingsFromLs = localStorage.getItem(getStorageKey(account))
+    return settingsFromLs ? JSON.parse(settingsFromLs) : getDefaultSettings()
+  } catch (error) {
+    return getDefaultSettings()
+  }
+}
+
+export const setSettings = (account: string, newSettings: UserSettings) => {
+  localStorage.setItem(getStorageKey(account), JSON.stringify(newSettings))
```

Comment:
The marginal naming difference between `setSetting` & `setSettings` rustles my jimmies a bit.

Does this need to be broken into two functions? If yes maybe this one could be renamed `setSettings` > `storeSettingToLs`?

Reply:
You know, I agree with your jimmies in this case. I removed `setSetting` all together. `setSettings` can easily do the work of both.